kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/EnvelopeEncryption.java

```diff
@@ -53,8 +53,12 @@ public EnvelopeEncryptionFilter<K> createFilter(FilterFactoryContext context, Co
         KmsService<Object, K, E> kmsPlugin = context.pluginInstance(KmsService.class, configuration.kms());
         Kms<K, E> kms = kmsPlugin.buildKms(configuration.kmsConfig());
 
-        var keyManager = new InBandKeyManager<>(kms, BufferPool.allocating(), 500_000, context.eventLoop(),
-                new ExponentialJitterBackoffStrategy(Duration.ofMillis(500), Duration.ofSeconds(5), 2d, ThreadLocalRandom.current()));
+        ExponentialJitterBackoffStrategy backoffStrategy = new ExponentialJitterBackoffStrategy(Duration.ofMillis(500), Duration.ofSeconds(5), 2d,
+                ThreadLocalRandom.current());
+        Kms<K, E> resilientKms = ResilientKms.get(kms, context.eventLoop(), backoffStrategy, 3);
+        Kms<K, E> resilientCachingKms = CachingKms.caching(resilientKms, 1000, Duration.ofHours(1), 1000, Duration.ofMinutes(10), Duration.ofMinutes(8));
```

Comment:
I think the cache sizes/expiration times ought to be driven by configuration with a fallback to the sensible defaults (your values seem okay to me).